core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -511,53 +514,48 @@ private ExecutorHolder rebuild(String name, ExecutorHolder previousExecutorHolde
 
             int size = applyHardSizeLimit(name, settings.getAsInt("size", defaultSize));
             SizeValue queueSize = getAsSizeOrUnbounded(settings, "capacity", getAsSizeOrUnbounded(settings, "queue", getAsSizeOrUnbounded(settings, "queue_size", defaultQueueSize)));
-            logger.debug("creating thread_pool [{}], type [{}], size [{}], queue_size [{}]", name, type, size, queueSize);
+            logger.debug("creating thread pool [{}], type [{}], size [{}], queue_size [{}]", name, type, size, queueSize);
             Executor executor = EsExecutors.newFixed(name, size, queueSize == null ? -1 : (int) queueSize.singles(), threadFactory, threadContext);
             return new ExecutorHolder(executor, new Info(name, threadPoolType, size, size, null, queueSize));
         } else if (ThreadPoolType.SCALING == threadPoolType) {
-            TimeValue defaultKeepAlive = defaultSettings.getAsTime("keep_alive", timeValueMinutes(5));
-            int defaultMin = defaultSettings.getAsInt("min", 1);
-            int defaultSize = defaultSettings.getAsInt("size", EsExecutors.boundedNumberOfProcessors(settings));
+
+            final Integer queueSize = settings.getAsInt("queue_size", null);
+            if (queueSize != null) {
+                throw new IllegalArgumentException("thread pool [" + name + "] can not have its queue re-sized but was [" + queueSize + "]");
+            }
+
+            final TimeValue keepAlive = settings.getAsTime("keep_alive", defaultSettings.getAsTime("keep_alive", null));
+            assert keepAlive != null;
+            if (keepAlive.millis() <= 0) {
+                throw new IllegalArgumentException("thread pool [" + name + "] must have positive keep_alive");
+            }
+
+            final Integer min = settings.getAsInt("min", defaultSettings.getAsInt("min", null));
+            assert min != null;
+            final Integer max = settings.getAsInt("max", defaultSettings.getAsInt("max", null));
```

Comment:
@clintongormley I discussed this with @ywelsch. The plan is the following: we are going to migrate the thread pool settings to the settings infrastructure. At this time I will change the name of this setting from `size` to `max` since `max` is a more appropriate name for it.